### DISCUSSION TITLE
niceGUI with PySerial?

### DISCUSSION THREAD

**Discussion by StroemJaeger** [zauberzeug/nicegui#2309] [Q&A]
### Question

Hi, i am excited about the ease of use of niceGUI. However, i miss an example demonstrating niceGUI operating in conjunction with data-exchange via a physical serial port. In post #334 i read that this was one of the main goals developing niceGUI.
In my case, i want to read out several energy-meters via ModBusRTU and an edge-device for representing the actual values to the user(s). Based on user-input and calculations in the edge-device  some output-units shall be controlled over the same ModBusRTU line. Is there any example available getting me to understand the basic principles?

**Comment by falkoschindler** [comment]
You're right, @StroemJaeger, a pyserial example is still missing. I just created PR #2320. What do you think?
